### ISSUE TITLE
Expose reports endpoints via REST

### ISSUE THREAD

**Issue by DrewAPicture** [easydigitaldownloads/easy-digital-downloads#6281]
Tracking issue: #6236
Related: #6280 

Ideally, it should be possible to make endpoints consumable via REST. This is not a must-have and is lower priority.

**Comment by cklosowski** [comment]
With reports in 3.0 now, I think it's best if we let that bake a bit before we start extending it all out.
